.kokoro/presubmit/downstream-compatibility.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+# Comma-delimited list of repos to test with the local java-shared-dependencies
+if [ -z "${REPOS_UNDER_TEST}" ]; then
+  echo "REPOS_UNDER_TEST must be set to run downstream-compatibility.sh"
+  exit 1
+fi
+
+# Get the directory of the build script
+scriptDir=$(realpath "$(dirname "${BASH_SOURCE[0]}")")
+cd "${scriptDir}/../.." # cd to the root of this repo
+source "$scriptDir/common.sh"
+
+echo "Setup maven mirror"
+mkdir -p "${HOME}/.m2"
```

Comment:
I might be missing something, I thought `.m2` folders would be created automatically by running `mvn` command?

Reply:
Likely, however we haven't yet invoked mvn at this point.

Reply:
What's the content in `settings.xml`? If we don't push to anywhere, I think we might be able to just use the default?

Reply:
https://github.com/googleapis/sdk-platform-java/blob/main/settings.xml

It uses the Google mirror instead of maven central directly.

This logic is not new to this PR. If we remove it from here, we should remove use of the mirror in all the repos that use it. I'd ask this discussion not be considered part of this PR.

https://github.com/googleapis/java-shared-config/blob/763419aefd7efb40b30641641198ca6491b380d2/.kokoro/presubmit/downstream-build.sh#L42-L44

https://github.com/googleapis/sdk-platform-java/blob/33e40ff788beaca80f2158f0d01642d9bc3d3c8f/.kokoro/presubmit/downstream-build.sh#L28-L30

https://github.com/googleapis/google-cloud-java/blob/a082d632ddb3aa413886b731e0b0298f2bec8170/.kokoro/build.sh#L26-L28

Reply:
> This logic is not new to this PR. If we remove it from here, we should remove use of the mirror in all the repos that use it. I'd ask this discussion not be considered part of this PR.

Agreed. It was mostly for my own understanding, @suztomo what's the benefit of using Google mirror? I guess it's more stable than Maven central when downloading dependencies locally?

Reply:
It's also faster in Kokoro CI because both are in Google'a infrastructure.